diff --git a/setup.py b/setup.py
index 77b0253..926b355 100755
--- a/setup.py
+++ b/setup.py
@@ -70,7 +70,7 @@ setup(
     keywords="csp constraint solving problems problem solver",
     # You can just specify the packages manually here if your project is
     # simple. Or you can use find_packages().
-    packages=find_packages(exclude=["contrib", "docs", "tests*"]),
+    packages=find_packages(exclude=["contrib", "docs", "examples", "tests*"]),
     # List run-time dependencies here.  These will be installed by pip when your
     # project is installed. For an analysis of "install_requires" vs pip's
     # requirements files see:
