diff --git a/xc/xc7/yosys/synth.tcl b/xc/xc7/yosys/synth.tcl
index 428ea9a1..f390d2fb 100644
--- a/xc/xc7/yosys/synth.tcl
+++ b/xc/xc7/yosys/synth.tcl
@@ -1,7 +1,7 @@
 yosys -import
 
-plugin -i xdc
-plugin -i fasm
+plugin -i $::env(YOSYS_SYMBIFLOW_PLUGINS)/xdc.so -a xdc
+plugin -i $::env(YOSYS_SYMBIFLOW_PLUGINS)/fasm.so -a fasm
 
 # Import the commands from the plugins to the tcl interpreter
 yosys -import
